File: geojson_to_kuro.py
from datetime import datetime
import json

class Shop:
    def __init__(self, name, latitude, longitude, address, products = []):
        self.name = name
        self.latitude = latitude
        self.longitude = longitude
        self.products = products
        self.address = address
    def __json__(self):
        return {
            'name': self.name,
            'latitude': self.latitude,
            'longitude': self.longitude,
            'products': self.products,
            'address': self.address.__json__()
        }
    for_json = __json__

class Address:
    def __init__(self, postal_code, street, house_number, city):
        self.postal_code = postal_code
        self.address = self.__build_address__(street, house_number)
        self.city = city
    def __build_address__(self, street = '', house_number = ''):
        if(type(street) == type(None)):
            street = ''
        return str(street) + ' ' + str(house_number)
    def __json__(self):
        return {
            'postalCode': self.postal_code,
            'address': self.address,
            'address2': '',
            'city': self.city,
            'district': '',
        }
    for_json = __json__

result = []
postal_codes = []
cities = []
start = datetime.now()
print('Started: ', start)
with open('export.geojson') as file:
    shop_data = json.load(file)
    
    for shop in shop_data['features']:
        
        try:
            shop_name = shop['properties']['name']
        except(KeyError):
            shop_name = ''

        shop_type = shop['geometry']['type']

        if (shop_type == 'Point'):
            shop_long = shop['geometry']['coordinates'][0]
            shop_lat = shop['geometry']['coordinates'][1]
        elif(shop_type == 'LineString'):
            shop_long = shop['geometry']['coordinates'][0][0]
            shop_lat = shop['geometry']['coordinates'][0][1]
        elif(shop_type == 'Polygon'):
            shop_long = shop['geometry']['coordinates'][0][0][0]
            shop_lat = shop['geometry']['coordinates'][0][1][1]
        elif(shop_type == 'MultiPolygon'):
            shop_long = shop['geometry']['coordinates'][0][0][0][0]
            shop_lat = shop['geometry']['coordinates'][0][0][0][1]
        else:
            print('Corrupt shop', shop_type)
            
        shop_address_postalcode = None
        shop_address_street = None
        shop_address_town = None
        shop_address_housenumber = None

        try:
            shop_address_postalcode = shop['properties']['addr:postcode']
            if(shop_address_postalcode not in postal_codes):
                postal_codes.append(shop_address_postalcode)
        except(KeyError):
            pass
        try:
            shop_address_street = shop['properties']['addr:street']
        except(KeyError):
            pass
        try:
            shop_address_town = shop['properties']['addr:city']
            if(shop_address_town not in cities):
                cities.append(shop_address_town)
        except(KeyError):
            pass
        try:
            shop_address_housenumber = shop['properties']['addr:housenumber']
        except(KeyError):
            pass
        sa = Address(
            shop_address_postalcode,
            shop_address_street,
            shop_address_housenumber,
            shop_address_town,
        )
        se = Shop(shop_name, shop_lat, shop_long, sa)
        
        result.append(se.__json__())
    
with open("./result.json", "w") as f:
    output = json.dumps(result, ensure_ascii=False)
    f.write(output)
    f.close()
end = datetime.now()
print('Ended: ', end)   
print('The task lasted: ', (end-start))
print(
    'In total this were ' + str(len(result)) + ' Shops to parse for our data.\n' +
    'The Shops are listed in: ' + str(len(postal_codes)) + ' postal codes.\n'
    'The Shops are listed in: ' + str(len(cities)) + ' cities.'

)

File: geojson_to_kuro.rb
#!/usr/bin/env jruby
require 'json'

result = []

class Shopentry
    def initialize(name, lon, lat, addr)
        @name = name
        @latitude = lat
        @longitude = lon
        @products = []
        @address = addr
    end

    def to_s
        s = StringIO.new
        s << @name
        s << " "
        s << @latitude
        s << " "
        s << @longitude
        s << " "
        s << @address

        s.string
    end

    def to_hash
        {
            name: @name,
            latitude: @latitude,
            longitude: @longitude,
            products: [],
            address: @address
        }
        end

    def to_json(options = {})
        to_hash.to_json
    end
end

class Address
    def initialize(postalCode, street, town, housenumber)
        s = StringIO.new
        s << street
        s << " "
        s << housenumber
        @postalCode = postalCode
        @address = s.string
        @address2 = ""
        @district = ""
        @city = town
    end

    def to_s
        s = StringIO.new
        s << @address
        s << " "
        s << @city
        
        s.string
    end

    def to_hash
        {
            postalCode: @postalCode,
            address: @address,
            address2: "",
            district: "",
            city: @city
        }
    end
    
    def to_json(options = {})
        to_hash.to_json
    end
end

t1 = Time.now
puts "Started at: " << t1.to_s

file = File.open("./export.geojson")
file_data = file.read

json_data = JSON.parse(file_data)

for obj in json_data['features'] do
    
    shop_name = obj['properties']['name']

    if(shop_name == nil) 
        next;
    end

    shop_lat = nil
    shop_lon = nil

    case obj['geometry']['type']
    when "Point"
        shop_lat = obj['geometry']['coordinates'][0]
        shop_lon = obj['geometry']['coordinates'][1]
    when "LineString"
        shop_lat = obj['geometry']['coordinates'][0][0]
        shop_lon = obj['geometry']['coordinates'][0][1]
    when "Polygon"
        shop_lat = obj['geometry']['coordinates'][0][0][0]
        shop_lon = obj['geometry']['coordinates'][0][0][1]
    when "MultiPolygon"
        shop_lat = obj['geometry']['coordinates'][0][0][0][0]
        shop_lon = obj['geometry']['coordinates'][0][0][0][1]
    end 

    shop_address_postalcode = obj['properties']['addr:postcode']
    shop_address_street = obj['properties']['addr:street']
    shop_address_town = obj['properties']['addr:city']
    shop_address_housenumber = obj['properties']['addr:housenumber']

    sa = Address.new(shop_address_postalcode, shop_address_street, shop_address_town, shop_address_housenumber)
    se = Shopentry.new(shop_name, shop_lat, shop_lon, sa)

    result << se;
end

f = File.new("./result.json", "w")
str = JSON.dump(result)
f.write(str)
f.close
t2 = Time.now
puts "Ended at: " << t2.to_s
puts "It took summa sumarum: " << (t2-t1).to_s << " ms."